Error 500 export results to PDF

Hello.
Deploy mobileAudit version 2.2.1

After scan i press button @Export@ get error 500.

And errros in logs:

mobileaudit-web-1       | [ERROR] 16/Dec/2022 08:55:24 - Internal Server Error: /export/1
mobileaudit-web-1       | Traceback (most recent call last):
mobileaudit-web-1       |   File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
mobileaudit-web-1       |     response = get_response(request)
mobileaudit-web-1       |   File "/usr/local/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
mobileaudit-web-1       |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
mobileaudit-web-1       |   File "/usr/local/lib/python3.9/contextlib.py", line 79, in inner
mobileaudit-web-1       |     return func(*args, **kwds)
mobileaudit-web-1       |   File "/usr/local/lib/python3.9/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
mobileaudit-web-1       |     return view_func(request, *args, **kwargs)
mobileaudit-web-1       |   File "./app/views.py", line 463, in export
mobileaudit-web-1       |     pdf = pdfkit.from_string(html, False, options)
mobileaudit-web-1       |   File "/usr/local/lib/python3.9/site-packages/pdfkit/api.py", line 72, in from_string
mobileaudit-web-1       |     return r.to_pdf(output_path)
mobileaudit-web-1       |   File "/usr/local/lib/python3.9/site-packages/pdfkit/pdfkit.py", line 159, in to_pdf
mobileaudit-web-1       |     raise IOError("wkhtmltopdf exited with non-zero code {0}. error:\n{1}".format(exit_code, stderr))
mobileaudit-web-1       | OSError: wkhtmltopdf exited with non-zero code 127. error:
mobileaudit-web-1       | /usr/bin/wkhtmltopdf: error while loading shared libraries: libQt5Core.so.5: cannot open shared object file: No such file or directory
mobileaudit-web-1       |
mobileaudit-web-1       | [pid: 13|app: 0|req: 54/75] 192.168.144.4 () {54 vars in 1220 bytes} [Fri Dec 16 08:55:18 2022] GET /export/1 => generated 145 bytes in 6303 msecs (HTTP/1.0 500) 7 headers in 217 bytes (1 switches on core 0)

--- a/Dockerfile
+++ b/Dockerfile
@@ -12,6 +12,9 @@ RUN apt-get update && \
 	apt-get clean && \
 	update-ca-certificates -f
 
+# Avoid issue in libqt5gui5 #73
+RUN strip --remove-section=.note.ABI-tag /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
+
 # Get JADX Tool
 ENV JADX_VERSION 1.4.5
 

--- a/app/templates/app.html
+++ b/app/templates/app.html
@@ -63,7 +63,7 @@
             {% else %}
             <td> {% fa5_icon 'shield-alt' color='green' %} {{ scan_data.antivirus.malicious }}</td>
             {% endif %} {% endif %}
-            <td><a href="{% url 'findings' scan_id=scan.id %}" target="_blank">{{ scan.findings }}</a></td>
+            <td><a href="{% url 'findings' scan_id=scan.id %}" target="_blank" rel="noopener noreferrer">{{ scan.findings }}</a></td>
             <td>
                 <table class="table table-borderless">
                     {% for severity, number in scan_data.findings.items %}

--- a/app/templates/export.html
+++ b/app/templates/export.html
@@ -506,17 +506,17 @@
                                 <tbody>
                                     {% for finding in findings_ordered|lookup:id %}
                                     <tr>
-                                        <td><a href="{% url 'finding' id=finding.id %}" target="_blank">{{ finding.id }}</a></td>
+                                        <td><a href="{% url 'finding' id=finding.id %}" target="_blank" rel="noopener noreferrer">{{ finding.id }}</a></td>
                                         <td>{{ finding.get_severity_display }}</td>
-                                        <td><a href="{% url 'finding_view_file' id=finding.id %}" target="_blank">{{ finding.path }}</a></td>
-                                        <td><a href="{% url 'finding_view_file' id=finding.id %}#finding" target="_blank">{{ finding.line_number }}</a></td>
+                                        <td><a href="{% url 'finding_view_file' id=finding.id %}" target="_blank" rel="noopener noreferrer">{{ finding.path }}</a></td>
+                                        <td><a href="{% url 'finding_view_file' id=finding.id %}#finding" target="_blank" rel="noopener noreferrer">{{ finding.line_number }}</a></td>
                                         <td>{{ finding.line }}</td>
                                         <td>{{ finding.get_status_display }}</td>
-                                        <td><a href="{{ settings.CWE_URL }}{{ finding.cwe.cwe }}.html" target="_blank">{{ finding.cwe.cwe }} </a></td>
+                                        <td><a href="{{ settings.CWE_URL }}{{ finding.cwe.cwe }}.html" target="_blank" rel="noopener noreferrer">{{ finding.cwe.cwe }} </a></td>
                                         {% if settings.DEFECTDOJO_ENABLED %}
                                         <td>
                                             {% if finding.defectdojo_id > 0 %}
-                                            <a href="{{ settings.DEFECTDOJO_URL }}{{ finding.defectdojo_id }}" target="_blank">{{ finding.defectdojo_id }}</a> {% else %} <span>N/A</span>{% endif %}
+                                            <a href="{{ settings.DEFECTDOJO_URL }}{{ finding.defectdojo_id }}" target="_blank" rel="noopener noreferrer">{{ finding.defectdojo_id }}</a> {% else %} <span>N/A</span>{% endif %}
                                         </td>
                                         {% endif %}
                                     </tr>
@@ -563,9 +563,9 @@
                             <tbody>
                                 {% for finding in best_practices %} {% if finding.type.id == practice.id %}
                                 <tr>
-                                    <td><a href="{% url 'finding' id=finding.id %}" target="_blank">{{ finding.id }}</a></td>
-                                    <td><a href="{% url 'finding_view_file' id=finding.id %}" target="_blank">{{ finding.path }}</a></td>
-                                    <td><a href="{% url 'finding_view_file' id=finding.id %}#finding" target="_blank">{{ finding.line_number }}</a></td>
+                                    <td><a href="{% url 'finding' id=finding.id %}" target="_blank" rel="noopener noreferrer">{{ finding.id }}</a></td>
+                                    <td><a href="{% url 'finding_view_file' id=finding.id %}" target="_blank" rel="noopener noreferrer">{{ finding.path }}</a></td>
+                                    <td><a href="{% url 'finding_view_file' id=finding.id %}#finding" target="_blank" rel="noopener noreferrer">{{ finding.line_number }}</a></td>
                                     <td>{{ finding.line }}
                                     </td>
                                 </tr>
@@ -601,7 +601,7 @@
                     {% if file.type == 'html' %}
                     <td>{{ file.name }}</td>
                     {% else %}
-                    <td><a href="{% url 'view_file' id=file.id %}" target="_blank">{{ file.name }} </a></td>
+                    <td><a href="{% url 'view_file' id=file.id %}" target="_blank" rel="noopener noreferrer">{{ file.name }} </a></td>
                     {% endif %}
                     <td>{{ file.type }} </td>
                 </tr>
@@ -632,7 +632,7 @@
                     <td>{{ string.id }}</td>
                     <td>{{ string.type }} </td>
                     <td>{{ string.value }} </td>
-                    <td><a href="{% url 'finding' id=string.finding.id %}" target="_blank">{{ string.finding.id }}</a></td>
+                    <td><a href="{% url 'finding' id=string.finding.id %}" target="_blank" rel="noopener noreferrer">{{ string.finding.id }}</a></td>
                 </tr>
                 {% endfor %}
             </tbody>

--- a/app/templates/finding.html
+++ b/app/templates/finding.html
@@ -15,11 +15,11 @@
     </tr>
     <tr>
         <th>CWE</th>
-        <td><a class="link" href="{{ settings.CWE_URL }}{{finding.cwe.cwe}}.html" target="_blank">{{ finding.cwe.cwe }} </a></td>
+        <td><a class="link" href="{{ settings.CWE_URL }}{{finding.cwe.cwe}}.html" target="_blank" rel="noopener noreferrer">{{ finding.cwe.cwe }} </a></td>
     </tr>
     <tr>
         <th>OWASP TOP 10 Mobile Risk</th>
-        <td><a class="link" href="{{ pattern.default_risk.reference }}" target="_blank">M{{ pattern.default_risk.risk }} </a></td>
+        <td><a class="link" href="{{ pattern.default_risk.reference }}" target="_blank" rel="noopener noreferrer">M{{ pattern.default_risk.risk }} </a></td>
     </tr>
     <tr>
         <th>Finding</th>
@@ -55,6 +55,6 @@
     </tr>
 </table>
 {% if settings.DEFECTDOJO_ENABLED and finding.defectdojo_id > 0 %}
-<a href="{{ settings.DEFECTDOJO_URL }}{{ finding.defectdojo_id }}" target="_blank" class="btn btn-outline-dark">DefectDojo{% fa5_icon 'external-link-alt' %}</a> {% endif %}
+<a href="{{ settings.DEFECTDOJO_URL }}{{ finding.defectdojo_id }}" target="_blank" rel="noopener noreferrer" class="btn btn-outline-dark">DefectDojo{% fa5_icon 'external-link-alt' %}</a> {% endif %}
 
 <a class="btn btn-outline-danger" href="{% url 'edit_finding' id=finding.id %}">Edit</a> {% endblock %}

--- a/app/templates/findings.html
+++ b/app/templates/findings.html
@@ -57,19 +57,19 @@
             {% for finding in findings %}
             <tr>
                 <td><input id="{{ finding.id }}" class="finding" name="{{ finding.id }}" type="checkbox" /></td>
-                <td><a class="btn btn-outline-success btn-round" href="{% url 'finding' id=finding.id %}" target="_blank">{{ finding.id }}</a></td>
+                <td><a class="btn btn-outline-success btn-round" href="{% url 'finding' id=finding.id %}" target="_blank" rel="noopener noreferrer">{{ finding.id }}</a></td>
                 <td>{{ finding.name }}</td>
                 <td>{{ finding.get_severity_display }}</td>
-                <td><a class="link" href="{% url 'finding_view_file' id=finding.id %}" target="_blank">{{ finding.path }}</a></td>
-                <td><a class="link" href="{% url 'finding_view_file' id=finding.id %}#finding" target="_blank">{{ finding.line_number }}</a></td>
+                <td><a class="link" href="{% url 'finding_view_file' id=finding.id %}" target="_blank" rel="noopener noreferrer">{{ finding.path }}</a></td>
+                <td><a class="link" href="{% url 'finding_view_file' id=finding.id %}#finding" target="_blank" rel="noopener noreferrer">{{ finding.line_number }}</a></td>
                 <td>{{ finding.line }}</td>
                 <td>{{ finding.get_status_display }}</td>
-                <td><a class="link" href="{{ settings.CWE_URL }}{{ finding.cwe.cwe }}.html" target="_blank">{{ finding.cwe.cwe }} </a></td>
-                <td><a class="link" href="{{ pattern.default_risk.reference }}" target="_blank">M{{ pattern.default_risk.risk }} </a></td>
+                <td><a class="link" href="{{ settings.CWE_URL }}{{ finding.cwe.cwe }}.html" target="_blank" rel="noopener noreferrer">{{ finding.cwe.cwe }} </a></td>
+                <td><a class="link" href="{{ pattern.default_risk.reference }}" target="_blank" rel="noopener noreferrer">M{{ pattern.default_risk.risk }} </a></td>
                 {% if settings.DEFECTDOJO_ENABLED %}
                 <td>
                     {% if finding.defectdojo_id > 0 %}
-                    <a href="{{ settings.DEFECTDOJO_URL }}{{ finding.defectdojo_id }}" target="_blank">{{ finding.defectdojo_id }}</a> {% else %} <span>'N/A'</span>{% endif %}
+                    <a href="{{ settings.DEFECTDOJO_URL }}{{ finding.defectdojo_id }}" target="_blank" rel="noopener noreferrer">{{ finding.defectdojo_id }}</a> {% else %} <span>'N/A'</span>{% endif %}
                 </td>
                 {% endif %}
             </tr>

--- a/app/templates/home.html
+++ b/app/templates/home.html
@@ -67,7 +67,7 @@
                             {% else %}
                             <td> {% fa5_icon 'shield-alt' color='green' %} {{ scan_data.antivirus.malicious }}</td>
                             {% endif %} {% endif %}
-                            <td><a href="{% url 'findings' scan_id=scan.id %}" target="_blank">{{ scan.findings }}</a></td>
+                            <td><a href="{% url 'findings' scan_id=scan.id %}" target="_blank" rel="noopener noreferrer">{{ scan.findings }}</a></td>
                             <td>
                                 <table class="table table-borderless">
                                     {% for severity, number in scan_data.findings.items %}

--- a/app/templates/patterns.html
+++ b/app/templates/patterns.html
@@ -50,8 +50,8 @@
                 <td>No</td>
                 <td>{% fa5_icon 'exclamation-circle' color='red' %}</td>
                 {% endif %}
-                <td><a class="link" href="https://cwe.mitre.org/data/definitions/{{finding.cwe.cwe}}.html" target="_blank">{{ pattern.default_cwe.cwe }} </a></td>
-                <td><a class="link" href="{{ pattern.default_risk.reference }}" target="_blank">M{{ pattern.default_risk.risk }} </a></td>
+                <td><a class="link" href="https://cwe.mitre.org/data/definitions/{{finding.cwe.cwe}}.html" target="_blank" rel="noopener noreferrer">{{ pattern.default_cwe.cwe }} </a></td>
+                <td><a class="link" href="{{ pattern.default_risk.reference }}" target="_blank" rel="noopener noreferrer">M{{ pattern.default_risk.risk }} </a></td>
             </tr>
             {% endfor %}
         </tbody>

--- a/app/templates/scan.html
+++ b/app/templates/scan.html
@@ -161,7 +161,7 @@
                 <table class="table table-bordered">
                     <tr>
                         <th style="width: 30%;">Number of findings</th>
-                        <td><a href="{% url 'findings' scan_id=scan.id %}" target="_blank">{{ scan.findings }}</a></td>
+                        <td><a href="{% url 'findings' scan_id=scan.id %}" target="_blank" rel="noopener noreferrer">{{ scan.findings }}</a></td>
                     </tr>
                     <tr>
                         <th style="width: 30%;">By Severity</th>
@@ -361,7 +361,7 @@
                 <a name="antivirus"></a>
                 <h5 class="card-title">Virus Total Scan</h5>
                 <button class="btn btn-outline-primary" onclick="updateVirusTotal({{ scan.id }})">{% fa5_icon 'redo-alt' %} VT Reload </button> {% if antivirus_scan.link %}
-                <a href="{{ settings.VIRUSTOTAL_FILE_URL }}{{ antivirus_scan.identifier }}" target="_blank" class="btn btn-outline-dark">VT Link {% fa5_icon 'external-link-alt' %}</a> {% endif %}
+                <a href="{{ settings.VIRUSTOTAL_FILE_URL }}{{ antivirus_scan.identifier }}" target="_blank" rel="noopener noreferrer" class="btn btn-outline-dark">VT Link {% fa5_icon 'external-link-alt' %}</a> {% endif %}
                 <br>
                 <table class="table table-bordered">
                     <tr>
@@ -532,7 +532,7 @@
             <div class="card-body">
                 <a name="findings"></a>
                 <h5 class="card-title">Findings</h5>
-                <a href="{% url 'create_finding' scan_id=scan.id %}" target="_blank" class="btn btn-outline-info">New Finding</a>
+                <a href="{% url 'create_finding' scan_id=scan.id %}" target="_blank" rel="noopener noreferrer" class="btn btn-outline-info">New Finding</a>
                 <p>Number of findings: {{ scan.findings }}</p>
                 <form id="form-findings" action="{% url 'findings' %}" method="post" enctype="multipart/form-data">
                     <input type="hidden" name="scan" value="{{ scan.id }}">
@@ -614,18 +614,18 @@
                                                 {% for finding in findings_ordered|lookup:id %}
                                                 <tr>
                                                     <td><input id="{{ finding.id }}" class="finding select_finding_{{ category.id }}" name="{{ finding.id }}" type="checkbox" /></td>
-                                                    <td><a class="btn btn-outline-success btn-round"  href="{% url 'finding' id=finding.id %}" target="_blank">{{ finding.id }}</a></td>
+                                                    <td><a class="btn btn-outline-success btn-round"  href="{% url 'finding' id=finding.id %}" target="_blank" rel="noopener noreferrer">{{ finding.id }}</a></td>
                                                     <td>{{ finding.get_severity_display }}</td>
-                                                    <td><a class="link" href="{% url 'finding_view_file' id=finding.id %}" target="_blank">{{ finding.path }}</a></td>
-                                                    <td><a class="link" href="{% url 'finding_view_file' id=finding.id %}#finding" target="_blank">{{ finding.line_number }}</a></td>
+                                                    <td><a class="link" href="{% url 'finding_view_file' id=finding.id %}" target="_blank" rel="noopener noreferrer">{{ finding.path }}</a></td>
+                                                    <td><a class="link" href="{% url 'finding_view_file' id=finding.id %}#finding" target="_blank" rel="noopener noreferrer">{{ finding.line_number }}</a></td>
                                                     <td>{{ finding.line }}</td>
                                                     <td>{{ finding.get_status_display }}</td>
-                                                    <td><a class="link" href="{{ settings.CWE_URL }}{{ finding.cwe.cwe }}.html" target="_blank">{{ finding.cwe.cwe }} </a></td>
-                                                    <td><a class="link" href="{{ pattern.default_risk.reference }}" target="_blank">M{{ pattern.default_risk.risk }} </a></td>
+                                                    <td><a class="link" href="{{ settings.CWE_URL }}{{ finding.cwe.cwe }}.html" target="_blank" rel="noopener noreferrer">{{ finding.cwe.cwe }} </a></td>
+                                                    <td><a class="link" href="{{ pattern.default_risk.reference }}" target="_blank" rel="noopener noreferrer">M{{ pattern.default_risk.risk }} </a></td>
                                                     {% if settings.DEFECTDOJO_ENABLED %}
                                                     <td>
                                                         {% if finding.defectdojo_id > 0 %}
-                                                        <a class="link" href="{{ settings.DEFECTDOJO_URL }}{{ finding.defectdojo_id }}" target="_blank">{{ finding.defectdojo_id }}</a> {% else %} <span>N/A</span>{% endif %}
+                                                        <a class="link" href="{{ settings.DEFECTDOJO_URL }}{{ finding.defectdojo_id }}" target="_blank" rel="noopener noreferrer">{{ finding.defectdojo_id }}</a> {% else %} <span>N/A</span>{% endif %}
                                                     </td>
                                                     {% endif %}
                                                 </tr>
@@ -673,9 +673,9 @@
                                     <tbody>
                                         {% for finding in best_practices %} {% if finding.type.id == practice.id %}
                                         <tr>
-                                            <td><a class="btn btn-outline-success btn-round" href="{% url 'finding' id=finding.id %}" target="_blank">{{ finding.id }}</a></td>
-                                            <td><a class="link" href="{% url 'finding_view_file' id=finding.id %}" target="_blank">{{ finding.path }}</a></td>
-                                            <td><a class="link" href="{% url 'finding_view_file' id=finding.id %}#finding" target="_blank">{{ finding.line_number }}</a></td>
+                                            <td><a class="btn btn-outline-success btn-round" href="{% url 'finding' id=finding.id %}" target="_blank" rel="noopener noreferrer">{{ finding.id }}</a></td>
+                                            <td><a class="link" href="{% url 'finding_view_file' id=finding.id %}" target="_blank" rel="noopener noreferrer">{{ finding.path }}</a></td>
+                                            <td><a class="link" href="{% url 'finding_view_file' id=finding.id %}#finding" target="_blank" rel="noopener noreferrer">{{ finding.line_number }}</a></td>
                                             <td>{{ finding.line }}
                                             </td>
                                         </tr>
@@ -711,7 +711,7 @@
                             {% if file.type == 'html' %}
                             <td>{{ file.name }}</td>
                             {% else %}
-                            <td><a class="link" href="{% url 'view_file' id=file.id %}" target="_blank">{{ file.name }} </a></td>
+                            <td><a class="link" href="{% url 'view_file' id=file.id %}" target="_blank" rel="noopener noreferrer">{{ file.name }} </a></td>
                             {% endif %}
                             <td>{{ file.type }} </td>
                         </tr>
@@ -742,7 +742,7 @@
                             <td>{{ string.id }}</td>
                             <td>{{ string.type }} </td>
                             <td>{{ string.value }} </td>
-                            <td><a class="btn btn-outline-success btn-round" href="{% url 'finding' id=string.finding.id %}" target="_blank">{{ string.finding.id }}</a></td>
+                            <td><a class="btn btn-outline-success btn-round" href="{% url 'finding' id=string.finding.id %}" target="_blank" rel="noopener noreferrer">{{ string.finding.id }}</a></td>
                         </tr>
                         {% endfor %}
                     </tbody>

